v2/en/ann-1015008001rc1.md.tt2

```diff
@@ -0,0 +1,196 @@
+    * feature: [ngx.errlog](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/errlog.md#readme): enabled the FFI-based API for the stream subsystem. _Thanks Thibault Charbonnier for the patch._
+    * feature: `os.getenv` is now able to retrieve `env` directives values from within [init_by_lua*](https://github.com/openresty/lua-nginx-module#init_by_lua). _Thanks Thibault Charbonnier for the patch._
+    * change: we now require OpenResty's LuaJIT branch to work properly on ARM64 since we make use of the `thread.exdata` Lua API.
+    * change: increased stack level on subsystem violation error to expose the offending module. _Thanks Datong Sun for the patch._
+    * feature: implemented the `ndk.*` API with [FFI](http://luajit.org/ext_ffi.html).
+    * bugfix: [ngx.process](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/process.md#readme): [type()](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/process.md#type) didn't return 'master' in master process. _Thanks spacewander for the patch._
+    * bugfix: now we only allow a single version of [ngx_lua](https://github.com/openresty/lua-nginx-module#readme) and [ngx_stream_lua](https://github.com/openresty/stream-lua-nginx-module#readme). _Thanks spacewander for the patch._
+    * optimize: fixed misuses of Lua global variables in our Lua code (caught by the new version of the lj-releng tool).
+    * doc: [ngx.process](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/process.md#readme): fixed a typo in the spelling of 'privileged agent'. _Thanks spacewander for the patch._
+    * doc: [ngx.process](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/process.md#readme): updated the process type list. _Thanks spacewander for the patch._
+    * doc: [ngx.balancer](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/balancer.md#readme): fixed a typo in the sample code. _Thanks chronolaw for the patch._
+    * doc: [ngx.ssl](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/ssl.md#readme): documented the `get_tls1_version_str()` API function.
+    * doc: [ngx.semaphore](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/semaphore.md#readme): switch status to production ready.
+* upgraded LuaJIT to 2.1-20190228: https://github.com/openresty/luajit2/tags
+    * feature: implemented a new [table.isempty()](https://github.com/openresty/luajit2/blob/v2.1-agentzh/README) API. _Thanks OpenResty Inc. for supporting this work._
+    * feature: implemented a new [table.isarray()](https://github.com/openresty/luajit2/blob/v2.1-agentzh/README) API. _Thanks OpenResty Inc. for supporting this work._
+    * feature: implemented a new [table.nkeys()](https://github.com/openresty/luajit2/blob/v2.1-agentzh/README) API. _Thanks OpenResty Inc. for supporting this work._
```

Comment:
Now we could point to https://github.com/openresty/luajit2#tablenkeys directly in those links

Reply:
Good catch. Fixed now.